src/js/view/floating/floating-controller.ts

```diff
@@ -0,0 +1,371 @@
+import type Model from 'controller/model';
+import type Preview from 'view/preview';
+import type { FloatConfig } from 'controller/model';
+import FloatingDragUI from 'view/floating/floating-drag-ui';
+import { OS } from 'environment/environment';
+import { deviceIsLandscape } from 'utils/dom';
+import { isIframe } from 'utils/browser';
+import {
+    addClass,
+    removeClass
+} from 'utils/dom';
+import { STATE_IDLE, STATE_COMPLETE, STATE_ERROR } from 'events/events';
+import { isNumber } from 'utils/underscore';
+import {
+    style,
+} from 'utils/css';
+import viewsManager from 'view/utils/views-manager';
+import type { BoundsObj } from 'types/generic.type';
+import { getPlayerSizeStyles } from 'view/utils/player-size';
+
+const FLOATING_TOP_OFFSET = 62;
+
+let _floatingPlayer: HTMLElement | null = null;
+const fpContainer: {
+    floatingPlayer: HTMLElement | null
+} = {
+    floatingPlayer: _floatingPlayer
+};
+Object.defineProperty(fpContainer, 'floatingPlayer', {
+    get: () => _floatingPlayer,
+    set: (val) => {
+        if (val === _floatingPlayer) {
+            return;
+        }
+        _floatingPlayer = val;
+        const watchersToRun = watchers.concat([]);
```

Comment:
this covers an edge case where a player is set to always float, but upon attempting to startFloating it realizes there is another player already floating. if the player stops floating, it should immediately attempt to float once the underlying floating player changes

we also create a new array before running the logic so that if there are multiple players waiting to float, and one starts, the call to add the watcher for the remaining players does not get wiped out by the `length=0` call